core/pipeline/label.go

```diff
@@ -28,6 +29,25 @@ func EngineLabel(engineName string) Label {
 	}
 }
 
+func LoadClientLabels(engineVersion string) []Label {
+	labels := []Label{
+		{
+			Name:  "dagger.io/client.os",
+			Value: runtime.GOOS,
+		},
+		{
+			Name:  "dagger.io/client.arch",
+			Value: runtime.GOARCH,
+		},
+		{
+			Name:  "dagger.io/engine.version",
+			Value: engineVersion,
```

Comment:
This value should probably come from the server-side, in the same place that we set the engine name label:

https://github.com/dagger/dagger/blob/e5bfff60cdc52f625571c898f763f8a02d476e2c/engine/server/buildkitcontroller.go#L241